src/main/java/com/scalar/db/storage/common/checker/OperationChecker.java

```diff
@@ -0,0 +1,281 @@
+package com.scalar.db.storage.common.checker;
+
+import com.scalar.db.api.Delete;
+import com.scalar.db.api.Get;
+import com.scalar.db.api.Mutation;
+import com.scalar.db.api.Operation;
+import com.scalar.db.api.Put;
+import com.scalar.db.api.Scan;
+import com.scalar.db.api.Selection;
+import com.scalar.db.exception.storage.MultiPartitionException;
+import com.scalar.db.io.Key;
+import com.scalar.db.io.Value;
+import com.scalar.db.storage.common.metadata.TableMetadata;
+import com.scalar.db.storage.common.util.Utility;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public final class OperationChecker {
+  private OperationChecker() {}
+
+  public static void check(Get get, TableMetadata metadata) {
```

Comment:
Is there any particular reason for making the methods static?
I think it passes around `TableMetadata` to a pretty deep level so it would be a bit simpler to make it have a constructor and manage an instance variable. What do you think?
(also it's a bit more consistent with other checkers like ColumnChecker and ConditionChecker)

Reply:
Honestly, I don't have any big reason for that. I made the methods static because the original check methods in the `Utility` class were static. Okay, let's make it have a constructor and manage an instance variable.